app/src/frontend/building/edit-history/building-edit-summary.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { EditHistoryEntry } from '../models/edit-history-entry';
+import { arrayToDictionary, parseDate } from '../../helpers';
+import { dataFields } from '../../data_fields';
+import { CategoryEditSummary } from './category-edit-summary';
+
+import './building-edit-summary.css';
+
+interface BuildingEditSummaryProps {
+    historyEntry: EditHistoryEntry
+}
+
+function formatDate(dt: Date) {
+    return dt.toLocaleString(undefined, {
+        weekday: 'short',
+        year: 'numeric',
+        month: 'short',
+        day: 'numeric',
+        hour: '2-digit',
+        minute: '2-digit'
+    });
+}
+
+const BuildingEditSummary: React.FunctionComponent<BuildingEditSummaryProps> = props => {
+    const entriesWithMetadata = Object
+            .entries(props.historyEntry.forward_patch)
+            .map(([key, value]) => {
+                const info = dataFields[key] || {};
+                return {
+                    title: info.title || `Unknown field (${key})`,
+                    category: info.category || 'Unknown category',
+                    value: value,
+                    oldValue: props.historyEntry.reverse_patch && props.historyEntry.reverse_patch[key]
+                };
+            });
```

Comment:
Prepare metadata for displaying the edit history, using the global `dataFields` definition of categories/fields